### ISSUE TITLE
Separate metrics server and API server (different ports)

### ISSUE THREAD

**Issue by protolambda** [sigp/lighthouse#1411]
## Description

Separate metrics and api endpoints.
This would allow for:
- firewall customization just for metrics
- metrics before genesis event appears, as the API is not online yet, but metrics can be useful still.
- eth2stats to report more data before genesis state is known

## Version

Latest master, 9ae9df8, using official Docker image.

## Present Behaviour

Metrics are currently available under `:5052/metrics`

Configured as part of the API:
```
  --http --http-address=0.0.0.0 --http-port=5052
```

## Expected Behaviour

Metrics should be made available under another port. Other clients default to `:8080/metrics`.

Options something like this:
```
  --metrics --metrics-address=0.0.0.0 --metrics-port=8080
```

## Steps to resolve

Other clients provide options to configure API and metrics separately, and run two http servers to support that.

### API configuration option comparison

For comparison, this is what the other clients enable you to configure for API and metrics endpoints:

#### Nimbus

```
  --rpc --rpc-port={{beacon_rpc_port}} --rpc-address=0.0.0.0
  --metrics --metrics-port={{beacon_metrics_port}} --metrics-address=0.0.0.0
```

#### Prysm
```
  --monitoring-host=0.0.0.0 --monitoring-port={{beacon_metrics_port}}
  --rpc-host=0.0.0.0 --rpc-port={{beacon_grpc_port}}
  --grpc-gateway-host=0.0.0.0
  --grpc-gateway-port={{beacon_rpc_port}}
```

### Teku
```
  --rest-api-enabled=true --rest-api-interface=0.0.0.0 --rest-api-port={{beacon_rpc_port}}
  --metrics-enabled --metrics-interface=0.0.0.0 --metrics-port={{beacon_metrics_port}}
  --metrics-host-allowlist="*"
  --rest-api-host-allowlist="*"
```
(note that the allowlist option is not the same as the common http header cross-origin header thing).

**Comment by protolambda** [comment]
Extension of the idea: mirror the metrics endpoint options and server code in the validator client. So the validator can have its own metrics, tracked directly from the validator process. E.g. track signature times, how many duties it gets, and maybe some validator performance stats, etc. The Prysm validator client does it like this, and it works well.

This would also be very helpful for setting up alerts for testnet/mainnet validator incidents. cc @zedt3ster

**Comment by paulhauner** [comment]
Sounds reasonable. Shame to run two HTTP servers, but I doubt it's a significant overhead.

We can address this when we do the refactor to the standard eth2 HTTP API.

**Comment by paulhauner** [comment]
I am implementing this in #1569:

```
  --metrics --metrics-address=0.0.0.0 --metrics-port=8080
```

Thank you for the detailed issue and output from other clients @protolambda.

**Comment by paulhauner** [comment]
Resolved via #1569
